.github/workflows/build_staging.yml

```diff
@@ -0,0 +1,142 @@
+      - name: Typecheck
+        run: npx --no-install tsc --noEmit
+      - name: Check Lint
+        run: npm run lint
+      - name: Set Timezone to +8 (Ubuntu)
+        run: sudo timedatectl set-timezone Asia/Singapore
+      - name: Test
+        run: npm run test -- --coverage
+        env:
+          SENTRY_ORG: ${{ secrets.SENTRY_ORG }}
+          SENTRY_PROJECT: ${{ secrets.SENTRY_PROJECT }}
+          SENTRY_AUTH_TOKEN: ${{ secrets.SENTRY_AUTH_TOKEN }}
+          SENTRY_DSN: ${{ secrets.SENTRY_DSN }}
+          APP_BUILD_VERSION: ${{ github.run_number }}
+          APP_BINARY_VERSION: "dev"
+          DOMAIN_FORMAT: ${{ secrets.TEST_DOMAIN_FORMAT }}
+          PROJECT_ID: ${{ secrets.EXPO_TEST_PROJECT_ID }}
+  build_staging:
+    name: Build Staging
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: ${{ github.head_ref }}
+      - name: Fetch Tags
+        run: |
+          git fetch --prune --unshallow --tags -f
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 16.x
+      - uses: expo/expo-github-action@v7
+        with:
+          eas-version: latest
+          expo-version: latest
+          packager: npm
+          token: ${{ secrets.EXPO_TOKEN }}
+          eas-cache: true
+          expo-cache: true
+      - name: Cache Node Modules
+        uses: actions/cache@v2
+        env:
+          cache-name: cache-node-modules
+        with:
+          # npm cache files are stored in `~/.npm` on Linux/macOS
+          path: ~/.npm
+          key: ${{ runner.os }}-build-${{ env.cache-name }}-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-build-${{ env.cache-name }}-
+            ${{ runner.os }}-build-
+            ${{ runner.os }}-
+      - name: Install Packages
+        run: |
+          npm config set //npm.pkg.github.com/:_authToken $NPM_TOKEN
+          npm ci --legacy-peer-deps
+        env:
+          NPM_TOKEN: ${{ secrets.GH_TOKEN_SUPPLYALLY_BOT }}
+      - name: Get Latest Binary Version # Binary Version will be x.x.x based on the latest tag
+        id: latestBinaryVersion
+        run: |
+          # Release tag finds the latest tag in the tree branch - i.e. prod-x.x.x
+          RELEASE_TAG=$(echo $(git describe --tags --abbrev=0))
+          # Using param substitution, we output x.x.x instead
+          echo "::set-output name=version::${RELEASE_TAG#*-}"
+      - name: Echo Version Details
+        run: |
+          echo Build number is $GITHUB_RUN_NUMBER
+          echo Latest release is ${{ steps.latestBinaryVersion.outputs.version }}
+      - name: Setup eas json
+        env:
+          APP_BUILD_VERSION: ${{ github.run_number }}
+          APP_BINARY_VERSION: ${{ steps.latestBinaryVersion.outputs.version }}
+        run: |
+          .github/scripts/set-versions.sh staging $APP_BUILD_VERSION $APP_BINARY_VERSION
+          cat eas.json
+      - name: Eas Build
```

Comment:
when we build staging, do we also create the apk and ipa files? do we get to access them on the expo build dashboard?

Reply:
yea we can get everything from the ui https://expo.dev/accounts/supplyallytest/projects/rationally/builds

Reply:
oh nice, so technically we dont need to upload the staging builds on the repo bah, just the production

Reply:
think so too